src/index.ts

```diff
@@ -1,27 +1,32 @@
-import { TASK_STORAGE_SAVE, TASK_STORAGE_COMPARE } from "./constants";
+import { TASK_STORAGE_COMPARE, TASK_STORAGE_SAVE } from "./constants";
 
 import { extendConfig, task, types } from "hardhat/config";
-import { ActionType } from "hardhat/types";
+import { ActionType, HardhatRuntimeEnvironment } from "hardhat/types";
 
+import { TASK_COMPILE } from "hardhat/builtin-tasks/task-names";
 import { compareConfigExtender, mergeCompareArgs } from "./config";
 import { StorageLayout } from "./storage/storage-layout";
-import { TASK_COMPILE } from "hardhat/builtin-tasks/task-names";
 import { CompareArgs } from "./types";
 
 import fsExtra from "fs-extra";
 
 extendConfig(compareConfigExtender);
 
-const storageSave: ActionType<CompareArgs> = async (taskArgs, env) => {
-  mergeCompareArgs(env, taskArgs);
-
+const reCompileArtifacts = async (env: HardhatRuntimeEnvironment) => {
+  // TODO: Do we need to remove artifacts, if we are going to compile force???
```

Comment:
It is done so to delete all artifacts, because sometimes the hardhat compiler saves old builds, and this breaks the logic of the plugin. For this reason, we need to delete them manually